### ISSUE TITLE
Failed to open stream: Permission denied

### ISSUE THREAD

**Issue by JHeijting** [firefly-iii/firefly-iii#3239]
I am running Firefly III version 5.1.1 using HASSIO plugin (I didn't understand the installation instructions).

**Description**
Earlier I have successfully imported CSV files without a problem. Also I have not encountered any problem before. Really like Firefly III; very impressive application! Somehow it stopped working and I don't know why. Already tried /flush as found in other issues. Also tried restarting the application / database.

**Extra info**
Debug information generated at 2020-04-06 11:16:52 Europe/Amsterdam for Firefly III version **5.1.1**.

| Variable | Content |
| --- | --- |
| FF version | 5.1.1 |
| FF API version | 1.0.2 |
| Installation ID | b2c27d92-be90-5c10-8589-005df5b314e6 |
| App environment | local |
| App debug mode | true |
| App cache driver | file |
| App logging | debug, stdout |
| PHP version | 7.3.11-1\~deb10u1 |
| Display errors | Off |
| Session start | 2020-04-01 00:00:00 |
| Session end | 2020-04-30 23:59:59 |
| Session first | 2018-01-01 00:00:00 |
| Error reporting | ALL errors |
| Host | Linux |
| Interface | fpm-fcgi |
| UserID | 1 |
| Attempt at "en" | false |
| Attempt at "English" | false |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| DB drivers | mysql, sqlite |
| Current driver | mysql |
| Login provider |   |
| Trusted proxies (.env) |  |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.163 Safari/537.36 |
| Loaded extensions | Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, sodium, standard, cgi-fcgi, mysqlnd, PDO, xml, bcmath, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, intl, json, ldap, exif, mysqli, pdo_mysql, pdo_sqlite, Phar, posix, readline, shmop, SimpleXML, sockets, sqlite3, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, Zend OPcache |

Stack trace
```
#0 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(2, 'file_put_conten...', '/var/www/firefl...', 135, Array)
#1 /var/www/firefly/vendor/league/flysystem/src/Adapter/Local.php(135): file_put_contents('/data/firefly/u...', 'Datum;Omschrijv...', 2)
#2 /var/www/firefly/vendor/league/flysystem/src/Filesystem.php(102): League\Flysystem\Adapter\Local->write('at-55.data', 'Datum;Omschrijv...', Object(League\Flysystem\Config))
#3 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Filesystem/FilesystemAdapter.php(225): League\Flysystem\Filesystem->put('at-55.data', 'Datum;Omschrijv...', Object(League\Flysystem\Config))
#4 /var/www/firefly/app/Repositories/ImportJob/ImportJobRepository.php(447): Illuminate\Filesystem\FilesystemAdapter->put('at-55.data', 'Datum;Omschrijv...')
#5 /var/www/firefly/app/Http/Controllers/Import/JobConfigurationController.php(147): FireflyIII\Repositories\ImportJob\ImportJobRepository->storeFileUpload(Object(FireflyIII\Models\ImportJob), 'import_file', Object(Illuminate\Http\UploadedFile))
#6 [internal function]: FireflyIII\Http\Controllers\Import\JobConfigurationController->post(Object(Illuminate\Http\Request), Object(FireflyIII\Models\ImportJob))
#7 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): call_user_func_array(Array, Array)
#8 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\Routing\Controller->callAction('post', Array)
#9 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Routing/Route.php(219): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(FireflyIII\Http\Controllers\Import\JobConfigurationController), 'post')
#10 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Routing/Route.php(176): Illuminate\Routing\Route->runController()
#11 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Routing/Router.php(681): Illuminate\Routing\Route->run()
#12 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(130): Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#13 /var/www/firefly/app/Http/Controllers/Import/JobConfigurationController.php(57): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#14 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(153): FireflyIII\Http\Controllers\Import\JobConfigurationController->FireflyIII\Http\Controllers\Import\{closure}(Object(Illuminate\Http\Request), Object(Closure))
#15 /var/www/firefly/app/Http/Controllers/Controller.php(96): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#16 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(153): FireflyIII\Http\Controllers\Controller->FireflyIII\Http\Controllers\{closure}(Object(Illuminate\Http\Request), Object(Closure))
#17 /var/www/firefly/app/Http/Middleware/InterestingMessage.php(59): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#18 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): FireflyIII\Http\Middleware\InterestingMessage->handle(Object(Illuminate\Http\Request), Object(Closure))
#19 /var/www/firefly/app/Http/Middleware/Binder.php(78): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#20 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): FireflyIII\Http\Middleware\Binder->handle(Object(Illuminate\Http\Request), Object(Closure))
#21 /var/www/firefly/app/Http/Middleware/Range.php(58): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#22 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): FireflyIII\Http\Middleware\Range->handle(Object(Illuminate\Http\Request), Object(Closure))
#23 /var/www/firefly/vendor/jc5/google2fa-laravel/src/Middleware.php(29): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#24 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): PragmaRX\Google2FALaravel\Middleware->handle(Object(Illuminate\Http\Request), Object(Closure))
#25 /var/www/firefly/app/Http/Middleware/Authenticate.php(73): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#26 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): FireflyIII\Http\Middleware\Authenticate->handle(Object(Illuminate\Http\Request), Object(Closure))
#27 /var/www/firefly/vendor/laravel/passport/src/Http/Middleware/CreateFreshApiToken.php(50): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#28 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Laravel\Passport\Http\Middleware\CreateFreshApiToken->handle(Object(Illuminate\Http\Request), Object(Closure))
#29 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(76): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#30 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle(Object(Illuminate\Http\Request), Object(Closure))
#31 /var/www/firefly/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#32 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#33 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(56): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#34 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#35 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#36 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
#37 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(66): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#38 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
#39 /var/www/firefly/app/Http/Middleware/Sandstorm.php(54): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#40 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): FireflyIII\Http\Middleware\Sandstorm->handle(Object(Illuminate\Http\Request), Object(Closure))
#41 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(105): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#42 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Routing/Router.php(683): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#43 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Routing/Router.php(658): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#44 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Routing/Router.php(624): Illuminate\Routing\Router->runRoute(Object(Illuminate\Http\Request), Object(Illuminate\Routing\Route))
#45 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Routing/Router.php(613): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#46 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(170): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#47 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(130): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#48 /var/www/firefly/app/Http/Middleware/InstallationId.php(59): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#49 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): FireflyIII\Http\Middleware\InstallationId->handle(Object(Illuminate\Http\Request), Object(Closure))
#50 /var/www/firefly/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#51 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Fideloper\Proxy\TrustProxies->handle(Object(Illuminate\Http\Request), Object(Closure))
#52 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#53 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
#54 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#55 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
#56 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#57 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle(Object(Illuminate\Http\Request), Object(Closure))
#58 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php(63): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#59 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure))
#60 /var/www/firefly/app/Http/Middleware/SecureHeaders.php(50): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#61 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): FireflyIII\Http\Middleware\SecureHeaders->handle(Object(Illuminate\Http\Request), Object(Closure))
#62 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(105): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#63 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(145): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#64 /var/www/firefly/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(110): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#65 /var/www/firefly/public/index.php(76): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#66 {main}
```

**Comment by JC5** [comment]
All I can tell from the error is that there are permission errors. Make sure that Firefly III can write to /var/www/firefly/storage. I'm not sure how to do this with HASSIO.

**Comment by JHeijting** [comment]
Thank you for your fast respond. I am not sure how to do this either, but I will look into it.

**Comment by JHeijting** [comment]
I think it might be because of the names of the import files. My CSV import had spaces in the name. First, I deleted the spaces and placed underscores (_). This didn't work. Than I made the name a lot shorter (it was >20 characters). This helped!

Also I notice that my configuration file (json) is causing issues. I have tried the same trick by shortening the name and such, but when I upload a CSV and config file I will get the error. I do not get the error when I upload a short named CSV file without the config file.
